tasks/src/main/scala/dagr/tasks/vc/VarDictJava.scala

```diff
@@ -0,0 +1,148 @@
+  */
+object VarDictJava extends Configuration {
+  /** Config key for the VarDictJava directory.  This should be the root project directory. */
+  val RootDirConfigKey = "vardictjava.dir"
+
+  /** Path to the VarDictJava directory. */
+  lazy val RootDir = configure[Path](RootDirConfigKey)
+
+  /** Path to the VarDictJava install directory with the VarDictJava startup script. */
+  lazy val VarDictJava = RootDir.resolve("build/install/VarDict/bin/VarDict")
+
+  /** Path to the VarDictJava bin directory with a multitude of perl scripts. */
+  lazy val BinDir = RootDir.resolve("VarDict")
+
+  /** Pulls a sample name out of a BAM file. */
+  private[vc] def extractSampleName(bam: PathToBam) : String = {
+    val in = SamReaderFactory.make().open(bam)
+    yieldAndThen(in.getFileHeader.getReadGroups.iterator().next().getSample) { in.close() }
+  }
+}
+
+/** Runs VarDictJava to produce its intermediate tabular format. */
+private class VarDictJava(tumorBam: PathToBam,
+                          bed: FilePath,
+                          ref: PathToFasta,
+                          tumorName: String,
+                          minimumAf: Double = 0.01,
+                          includeNonPf: Boolean = false,
+                          minimumQuality: Int = 0,
+                          maximumMismatches: Int = 8,
+                          minThreads: Int = 1,
+                          maxThreads: Int = 32,
+                          memory: Memory = Memory("8G")
+                         ) extends ProcessTask with VariableResources with PipeOut[Vcf] {
+  name = "VarDictJava"
+
+  override def pickResources(resources: ResourceSet): Option[ResourceSet] = {
+    resources.subset(minCores = Cores(minThreads), maxCores = Cores(maxThreads), memory = memory)
+  }
+
+  override def args: Seq[Any] = {
+    val vdScript = VarDictJava.VarDictJava
+    val buffer   = new ListBuffer[Any]()
+
+    buffer.appendAll(List(vdScript, "-G", ref, "-N", tumorName, "-b", tumorBam))
+    buffer.appendAll(List("-z", "-c", "1", "-S", "2", "-E", "3", "-g", "4"))
+    buffer.append("-f", minimumAf)
+    if (!includeNonPf) buffer.append("-F", "0x700")
+    buffer.append("-q", minimumQuality)
+    buffer.append("-m", maximumMismatches)
+    buffer.append("-th", resources.cores.toInt)
+    buffer.append(bed)
+
+    buffer
+  }
+}
+
+/**
+  * Tumor-only single-sample variant calling with VarDictJava: https://github.com/AstraZeneca-NGS/VarDictJava
+  *
+  * The output Tumor name will be inferred from the the first read group in the input BAM if not provided.
+  */
+class VarDictJavaEndToEnd(tumorBam: PathToBam,
+                          bed: FilePath,
+                          ref: PathToFasta,
+                          out: PathToVcf = Io.StdOut,
+                          tumorName: Option[String] = None,
+                          minimumAf: Double = 0.01,
+                          includeNonPf: Boolean = false,
+                          minimumQuality: Int = 0,
```

Comment:
If these were `Options` you wouldn't have to repeat the hard-coded default here ;)